lib/spack/spack/cmd/license.py

```diff
@@ -89,50 +90,101 @@ def list_files(args):
         print(os.path.join(spack.paths.spack_root, relpath))
 
 
+class LicenseError(object):
+    OLD_LICENSE = 1
+    SPDX_MISMATCH = 2
+    GENERAL_MISMATCH = 3
```

Comment:
Can these be strings?  `int` constants in Python feel weird -- if you make them constant strings, the error messages are easy to generate from the strings.  Error_counts can be a `defaultdict(str)`

Reply:
I can change them to strings, but FYI I'm not sure if

> if you make them constant strings, the error messages are easy to generate from the strings

is true. If you are referring to the messages in `error_messages` that will require employing a method to align the values based on the max lengths of the strings (not hard but adds more logic).

Int constants in Python are one of the ways of implementing an enum, so IMO not weird, but again I can make them strings.

Let me know if you want them to be used to generate the error message.

Reply:
>  Error_counts can be a defaultdict(str)

The argument represents the instantiation type of the value, so even if the keys are strings, I think it would be `defaultdict(int)`, unless I'm missing something.

Reply:
Based on https://github.com/spack/spack/pull/11381#discussion_r281787796 I'm assuming this is not essential.